packages/netlify-cms-core/src/reducers/entryDraft.js

```diff
@@ -36,12 +36,9 @@ const entryDraftReducer = (state = Map(), action) => {
     case DRAFT_CREATE_FROM_ENTRY:
       // Existing Entry
       return state.withMutations(state => {
-        state.set('entry', action.payload.entry);
+        state.set('entry', fromJS(action.payload.entry));
         state.setIn(['entry', 'newRecord'], false);
-        // An existing entry may already have metadata. If we surfed away and back to its
-        // editor page, the metadata will have been fetched already, so we shouldn't
-        // clear it as to not break relation lists.
-        state.set('fieldsMetaData', action.payload.metadata || Map());
+        state.set('fieldsMetaData', Map());
         state.set('fieldsErrors', Map());
         state.set('hasChanged', false);
         state.set('key', uuid());
```

Comment:
I complained about this earlier, it looks like this `key` is causing all widgets to unmount and remount between calls. Could you please double-check.

Reply:
@barthc we does this when loading a new entry to ensure state is reset for all widgets. Is it causing an issue?

Reply:
Ok, got it, but before #3306 was merged the `relation` widget was kind of unusable, check out the gif below:

![Peek 2020-03-13 23-07](https://user-images.githubusercontent.com/8580864/76662913-bb8c2f80-657f-11ea-8999-0d5553b38a1b.gif)

The issue is that the unmount/mount is causing everything below this [line](https://github.com/netlify/netlify-cms/blob/master/packages/netlify-cms-widget-relation/src/RelationControl.js#L65) to always run.

Reply:
Ah - did not know about this, thanks for the gif. @erezrokah I know you were working in connection with relation widget and unsaved changes recently, any thoughts on this?

Reply:
@barthc fixed the unsaved changes bug for the relation widget.
I think the "original" relation widget bug was intensified by the unmount/mount behaviour to show up on each save instead of just initial load.

Reply:
Regardless, my change avoids it completely by creating the draft after loading the entry instead of when the component updates.